.github/workflows/run_tests.yml

```diff
@@ -0,0 +1,64 @@
+# This workflow will install Python dependencies, run tests,
+# and report test results and code coverage as artifacts. It will
+# be called by the workflow that run tests against new PRs and as
+# a first step in the workflow that publishes new Docker images.
+
+name: A reusable workflow to build and run the unit test suite
```

Comment:
I think I'll make a quick PR tomorrow doing the same thing for HGA - this is really nice. Thanks, @flamingbear!